ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py

```diff
@@ -0,0 +1,43 @@
+# matmul
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=ivy_np.valid_float_dtypes,
+        min_num_dims=3,
+        max_num_dims=5,
+        min_dim_size=2,
+        max_dim_size=5,
+    ),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.numpy.linalg.matmul"
+    ),
+    fm=st.booleans(),
+)
+
+def matmul(
+    x1: array,
+    x2: array,
+    /,
+    out=None,
+    *,
+    casting='same_kind',
+    order='K',
+    dtype=None,
+    subok=True
+):
+    helpers.frontend_function(

```

Comment:
this is where the implementation using ivy should be defined, you can refer [example](https://github.com/unifyai/ivy/blob/6f27c45a4deef5ec2ad1e5117ef2937ce6f99943/ivy/functional/frontends/numpy/mathematical_functions/arithmetic_operations.py#L68) and perform the changes accordingly

Reply:
Done